src/software/thunderscope/robot_communication.py

```diff
@@ -225,6 +227,21 @@ def __forward_to_proto_unix_io(self, type, data):
         if self.running:
             self.current_proto_unix_io.send_proto(type, data)
 
+    def __reduce_primitive_size(self, primitive):
+        """
+        Reduces the size of the primitive by removing the static obstacles
+        :param primitive: the primitive to be reduced
+        :return: The reduced primitive
+        """
+
+        # The static_obstacles array is the largest part of the Primitive proto.
+        # Since it is not used by the robots, we will remove all values from its
+        # array.
+        if primitive.HasField("move"):
+            del primitive.move.motion_control.static_obstacles[:]
```

Comment:
From testing, this didnt see to work (no diagnostics or play selected). Maybe try setting the static obstacles to an empty array?

Reply:
This is different than the initial field testing you did with the patch I sent. Have you tested this updated version as well?